### PR TITLE
ensure core_params are Registry object

### PR DESCRIPTION
**Pull request by fcoulter**
Pull Request for Issue #19487 . 

### Summary of Changes
adds test for whether the core_params are a string, if they are converts them to a Registry object

Without this fix, a fatal error can occur in a content plugin that uses the parameters, as for example in the jw_disqus plugin as described in the issue tracker

### Testing Instructions
install jw_disqus plugin

create a menu item of type tags component compact tag list

add this code to components/com_tags/views/tag/view.html.php

if (is_string($itemElement->core_params))
{
		$itemElement->core_params = new Registry($itemElement->core_params);
}

### Expected result
Without the added code, this causes a fatal error in Joomla 3.8.5

### Actual result

With the added code, the page renders as it should

### Documentation Changes Required

None that I can see

### CODE REVIEW

**Comment by brianteeman** [comment]
Please can you resolve the conflicts - i guess your branch was out of date

File: htaccess.txt [hunk 1]
```diff
@@ -68,6 +68,8 @@ RewriteRule .* - [E=HTTP_AUTHORIZATION:%{HTTP:Authorization}]
 # If the requested path and file is not /index.php and the request
 # has not already been internally rewritten to the index.php script
 RewriteCond %{REQUEST_URI} !^/index\.php
+# the request is not for an image file
+RewriteCond %{REQUEST_URI} !\.jpg|\.png|\.gif$
```
Comment (brianteeman) [inline htaccess.txt]: this change is unrelated to this PR please remove it

File: components/com_tags/views/tag/view.html.php [hunk 2]
```diff
@@ -91,6 +91,11 @@ public function display($tpl = null)
 
 				// For some plugins.
 				!empty($itemElement->core_body)? $itemElement->text = $itemElement->core_body : $itemElement->text = null;
+				if(is_string($itemElement->core_params))
```
Comment (Quy) [inline components/com_tags/views/tag/view.html.php]: Please add a blank line above line 94, Add a space after `if`.

File: components/com_tags/views/tag/view.html.php [hunk 3]
```diff
@@ -91,6 +91,11 @@ public function display($tpl = null)
...
+				{
+					$itemElement->core_params = new Registry($itemElement->core_params);
+				}
+
```
Comment (Quy) [inline components/com_tags/views/tag/view.html.php]: Remove blank line 98.

File: htaccess.txt (hunk 1 above)
Comment (fcoulter) [reply htaccess.txt]: Sorry, this was a mistake, should not have been there.

**Comment by laoneo** [comment]
When I see the change. Do we need to handle also null?

**Comment by fcoulter** [comment]
I am not sure why the file components/com_tags/views/tag/view.html.php is still showing as conflicting, so far as I can see it is the same as in the current version.

I am going to close this PR and start again, something has gone wrong.

**Comment by fcoulter** [comment]
Replaced by https://github.com/joomla/joomla-cms/pull/19684
